### ISSUE TITLE
Sort out styling of references in articles

### ISSUE THREAD

**Issue by bernhold** [betterscientificsoftware/bssw.io#1162]
We seem to be tending to use the footnote-style references more now that we have wikize_refs working well.  But way the references are styled could stand improvement.

Matt wants to use <div> tags to denote particular styling for references, but when these are used, anything within them is treated as HTML, whereas we're generating Markdown.  So that's one problem. (Matt's primarily wanting to change the font size and present them in a multi-column format.)

And even if we try to use the default styles with the generated Markdown for references, they look a little weird.  They are in an unordered list with the markers as superscripts.  It might be more natural, for example, to simply generate an ordered list with the markers as the labels for the list items.  Though it is more complicated because with wikize_refs the markers can actually be anything, it is only convention that says they're numbers.

**Comment by markcmiller86** [comment]
> Matt's primarily wanting to change the font size and present them in a multi-column format.

I believe `wikize_refs.py -t` or `wikize_refs.py` --two-column` does most of what @sbxchicago wants here now, doesn't it?

We *generate* markdown because that is the original source language and we'd like as much as possible to steer clear of anyone having to code `HTML` directly. There is a teeny, tiny bit of `HTML` for the `wikize_refs.py` citations to work to get superscripting in the markdown. Other markdowns support superscripting natively with caret, `^`, character which, honestly, I would prefer.

How the list of references is ultimately handled at the bottom of the documentation is I think entirely flexible because its *generated*. So, no content creaters need to be burdened with it.

**Comment by markcmiller86** [comment]
> So wikize_refs would have to reformat into HTML...

And, I thought it does that now with `-t` option, at least enough of the HTML wrapping that the back-end can pick it up and do the rest.

**Comment by markcmiller86** [comment]
For example, see these lines...

https://github.com/betterscientificsoftware/bssw.io/blob/8896daad588c112a2c64fbca02b645b6cedb97b7/utils/wikize_refs.py#L597-L610

**Comment by markcmiller86** [comment]
Ok, I am obviously misunderstanding things here. I thought the back-end went on to process all the markdown in a file including the markdown representing these refs. The HTML lines (beginning and ending tags) pass through unchanged so the final result is the html rendered by the backend bracketed by the styling tags `wikize_refs.py` inserted for the `divs` it creates.

**Comment by bernhold** [comment]
What you describe would amount to the Markdown processor passing individual HTML tags unaltered and interpreting everything else as markdown.  But actually anything *within* HTML tag pairs is treated as HTML and passed unaltered, along with the enclosing tags.

So `<div>[text](URL)</div>` would be rendered as `<div>[text](URL)</div>`. *not* `<div><a href="URL">text</a></div>`

**Comment by bernhold** [comment]
And to go back to the context of wikize_refs, to get the behavior you described, you'd need to isolate each text fragment within HTML tags and render that from markdown to HTML and assemble the whole thing so that within the div tags it is all HTML.  Which I think would be a lot more work than you're doing now.

**Comment by markcmiller86** [comment]
@rinku I don't think we can make forward progress on this any time soon. It requires a combination of things from our end and from the backend processor.
